src/drivers/rp-pwm.adb

```diff
@@ -219,6 +220,28 @@ package body RP.PWM is
       end loop;
    end IRQ_Handler;
 
+   function Compare_Reg_Address (Pt : PWM_Point) return System.Address is
+      use System.Storage_Elements;
+
+      Base : constant System.Address := PWM_Periph.CH (Pt.Slice).CC'Address;
+   begin
+      case Pt.Channel is
+         when A =>
+            return Base;
+         when B =>
+            return To_Address (To_Integer (Base) + 2);
+      end case;
+
+      --  This is a bit of convoluted solution, but GNAT 11 seems to generate
+      --  invalid code with the following:
```

Comment:
I think these writes need to be 32-bit aligned, if you do `'Address` on the B register, it's on a 16-bit boundary. I'm not sure if it's safe to hand this to the DMA, will check the datasheet.

Reply:
That's a good point.

My code was "inspired" by https://github.com/adafruit/circuitpython/blob/6af4c7797c5df94aff24bece7797ddc46adbb42c/ports/raspberrypi/common-hal/audiopwmio/PWMAudioOut.c#L205

But I am not so sure about what it does now...

> If I understand the example code in that section correctly, it means a short write to an I/O register by the DMA controller will be "replicated", meaning that whatever value you write into the B register would be replicated in A as well.

That's my understanding as well. So I don't understand how the CircuitPython code deals with that. Maybe they just accepts that the data will be replicated...